alacritty/src/input/keyboard.rs

```diff
@@ -338,7 +338,15 @@ impl SequenceBuilder {
             let base_character = character.to_lowercase().next().unwrap();
 
             let codepoint = u32::from(character);
-            let base_codepoint = u32::from(base_character);
+            let mut base_codepoint = u32::from(base_character);
+
+            // Try to get the base for keys which change based on modifier, like `1` for `!`.
+            match key.key_without_modifiers().as_ref() {
+                Key::Character(unmodded) if codepoint == base_codepoint => {
```

Comment:
This might be cleaner with `matches!`?

Reply:
you can't use matches here from what I can see, since you need to `bind`.